### PR TITLE
Feat/grid_with_directions

### PR DESCRIPTION
**Pull request by MelchiorSchuh**

### CODE REVIEW

File: src/geode/geometry/square_matrix.cpp [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+                matrix_rows[d1].set_value( d2, value( d2, d1 ) );
+            }
+        }
+        return { std::move( matrix_rows ) };
+    }
+
+    template <>
+    SquareMatrix< 2 > SquareMatrix< 2 >::inverse() const
+    {
+        const auto det = determinant();
```
Comment (BotellaA) [inline src/geode/geometry/square_matrix.cpp:110]: det is not used?

File: src/geode/mesh/core/light_regular_grid.cpp [hunk 2]
```diff
@@ -38,12 +38,7 @@ namespace geode
         : public detail::GridImpl< dimension >
     {
     public:
-        Impl( Point< dimension > origin ) : origin_{ std::move( origin ) } {}
-
-        const Point< dimension >& origin() const
-        {
-            return origin_;
-        }
+        Impl() = default;
```
Comment (BotellaA) [inline src/geode/mesh/core/light_regular_grid.cpp:41]:
```suggestion

```

File: src/geode/geometry/coordinate_system.cpp [hunk 3]
```diff
@@ -152,6 +137,24 @@ namespace geode
         return { { -distance_AP / distance_A, distance_BP / distance_B } };
     }
 
+    template <>
+    Point< 3 > opengeode_geometry_api CoordinateSystem< 3 >::coordinates(
```
Comment (panquez) [inline src/geode/geometry/coordinate_system.cpp:141]: What about perf tests?

Comment (MelchiorSchuh) [reply src/geode/geometry/coordinate_system.cpp:141]: Ongoing

**Review by BotellaA — APPROVED** [review]

**Review by panquez — APPROVED** [review]

**Pull request merged**